src/cdomain/value/cdomains/valueDomain.ml

```diff
@@ -1003,26 +1003,29 @@ struct
           end
         | Blob (x,s,zeroinit), _ ->
           begin
-            let l', o' = shift_one_over l o in
-            let x = zero_init_calloced_memory zeroinit x t in
-            (* Strong update of scalar variable is possible if the variable is unique and size of written value matches size of blob being written to. *)
-            let do_strong_update =
-              begin match v with
-                | (Var var, _) ->
-                  let blob_size_opt = ID.to_int s in
-                  not @@ ask.is_multiple var
-                  && GobOption.exists (fun blob_size -> (* Size of blob is known *)
-                      (not @@ Cil.isVoidType t     (* Size of value is known *)
-                       && Z.equal blob_size (Z.of_int @@ Cil.alignOf_int t))
-                      || blob_destructive
-                    ) blob_size_opt
-                | _ -> false
-              end
-            in
-            if do_strong_update then
-              Blob ((do_update_offset ask x offs value exp l' o' v t), s, zeroinit)
-            else
-              mu (Blob (join x (do_update_offset ask x offs value exp l' o' v t), s, zeroinit))
+            match offs, value with
+            | `NoOffset, Blob (x2, s2, zeroinit2) -> mu (Blob (join x x2, ID.join s s2,ZeroInit.join zeroinit zeroinit2))
```

Comment:
The existing outer match has `offs` as the second value (which is ignored in this case). Before this case in the outer match, `Blob` with `Index` and `Field` are already handled, so it should be the case that `offs` will always be `NoOffset` here anyway.
That would get rid of my concern about `NoOffset` somehow being special here.

Reply:
That's true! I have simplified the code now!